src/vue/sheets/actor/character/TalentsTab.vue

```diff
@@ -88,6 +88,50 @@ async function upgradeTalent(talent: GenesysItem<TalentDataModel>) {
 	});
 }
 
+async function downgradeTalent(talent: GenesysItem<TalentDataModel>) {
+	if (talent.systemData.ranked === 'no') {
+		ui.notifications.info(game.i18n.format('Genesys.Notifications.TalentCannotBeDowngraded', { talentName: talent.name }));
+		return false;
+	}
+
+	const newEffectiveTier = talent.systemData.effectivePreviousTier;
+	const reward = talent.systemData.downgradeReward;
+
+	await toRaw(talent).update({
+		'system.rank': talent.systemData.rank - 1,
+	});
+
+	await toRaw(context.data.actor).update({
+		'system.experienceJournal.entries': [
+			...system.value.experienceJournal.entries,
+			{
+				amount: +reward,
+				type: EntryType.TalentRank,
+				data: {
+					name: talent.name,
+					id: talent.id,
+					tier: newEffectiveTier,
+					rank: talent.systemData.rank,
+				},
+			},
+		],
+	});
+}
+
+async function deleteTalent(talent: GenesysItem<TalentDataModel>) {
+	const reward = talent.systemData.refundReward;
+
+	await toRaw(context.data.actor).update({
+		'system.experienceJournal.entries': [
+			...system.value.experienceJournal.entries,
+			{
+				amount: +reward,
+			},
+		],
+	});
+	await toRaw(talent).delete()
+}
```

Comment:
I'm assuming that when you delete a talent it's because you made a mistake adding it and this feature just lets you delete the talent directly from the Talent tab instead of going to the Journal tab. Given that, creating a new entry seems like the wrong thing to do; instead we need to delete the entry that is associated to the deleted talent.

Reply:
I haven't noticed that you can delete talents and downgrade talents from journal tab. 
My bad.

The users are allowed to upgrade talents on talent tab and adding talent by drag and drop there, but to delete them you need to go to journal tab. This is counterintuitive.
I'll check journal tab code and try to reuse it in talent tab.